include/pipettes/core/communication.hpp

```diff
@@ -41,6 +41,15 @@ auto MessageReader::read(Reader &communication)
         case static_cast<uint32_t>(pipette_messages::MessageType::stop):
             r = pipette_messages::Stop{};
             break;
+        case static_cast<uint32_t>(pipette_messages::MessageType::setup):
```

Comment:
we should consider changing this code so that each switch case `return`s. while it can sometimes be good practice to maintain a single point of return, in a case like this removing the return from line 70 and having it in each switch case means compilation will fail if you forget one, whereas right now everything will work until runtime if you forget an `r =...` line.

Reply:
Am I fixing this correctly? Do I keep the `default` case the same or does it need to raise an error?

Reply:
this is fine although at some point we should probably put more thought into what happens when an unmatched message is received